scipy/stats/stats.py

```diff
@@ -8121,6 +8121,281 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
+    >>> print('Indexes:  ',lb, ub)
+    Indexes:   1 7
+    
+    >>> # Don't forgot to sort your data if working with indexes!
+    >>> print('CI bounds:',sorted(x)[lb], sorted(x)[ub])
+    CI bounds: 2 8
+
+    Generally, the more samples, the narrower the confidence intervals.
+
+    >>> for N in [5,10,100,1000]:
+    ...     x = np.random.rand(N)
+    ...     lb, ub = confint_quantile(x, 0.5, 0.95)
+    ...     med = np.median(sorted(x))
+    ...     print('N=%i'%N)
+    ...     print('lower-bound  %0.2f' % lb)
+    ...     print('emp. median  %0.2f' % med)
+    ...     print('upper-bound  %0.2f' % ub)
+    N=5
+    lower-bound  0.39  #random
+    emp. median  0.48  #random
+    upper-bound  0.96  #random
+    N=10
+    lower-bound  0.11  #random
+    emp. median  0.35  #random
+    upper-bound  0.71  #random
+    N=100
+    lower-bound  0.46  #random
+    emp. median  0.54  #random
+    upper-bound  0.63  #random
+    N=1000
+    lower-bound  0.47  #random
+    emp. median  0.52  #random
+    upper-bound  0.53  #random
+
+    .. versionadded:: 1.6.4
+    """
+
+    ##
+    # Checking the inputs
+    #
+    # x can be either an integer or a one-dimensional array-like
+    if isinstance(x, int):
+        if x < 1:
+            raise ValueError(
+                """Invalid data 'x': %s,
+            `x` must be either a strictly positive integer
+            or one-dimensional array-like.""" % repr(x))
+        n = x
+        # The function will returns the confint indexes
+        return_index = True
+    else:
+        x = np.asarray(x)
+        if x.ndim != 1:
+            raise ValueError(
+                """Invalid data 'x': %s,
+            `x` must be either a strictly positive integer
+            or one-dimensional array-like.""" % repr(x))
+        x = np.sort(x, axis=0)
+        n = x.shape[0]
+        # The function returns the confint as values of x
+        return_index = False
+    #
+    # `confidence` and `quantile` must be between 0 and 1
+    if confidence >= 1 or confidence <= 0:
+        raise ValueError(
+            """Invalid `confidence`: %s.
+            Provide a real number strictly between 0 and 1."""
+            % repr(confidence))
+    if quantile >= 1 or quantile <= 0:
+        raise ValueError(
+            """Invalid `quantile`: %s.
+            Provide a real number strictly between 0 and 1."""
+            % repr(quantile))
+    #
+    # `type` can be only `one-sided` or `two-sided`
+    if not (type == 'one-sided' or type == 'two-sided'):
+        raise ValueError(
+            """Invalid parameter: %s.
+            Valid 'type' values: 'one-sided' or 'two-sided'""" % repr(type))
+    ##
+
+    # Handle the type of intervals (one- or two-sided)
+    if type == 'two-sided':
+        conf_working = (1+confidence)/2
+    else:
+        # type == 'one-sided'
+        conf_working = confidence
+
+    # Compute the lower bound
+    LB = _confint_lowerbound(n, quantile, conf_working)
```

Comment:
Can we use lower-case `lb` and `ub` for the things that are returned, and something else for the temporary `lb` variable?